ch04/src/Solution423.java

```diff
@@ -0,0 +1,35 @@
+import java.util.Scanner;
+
+public class Solution423 {
+    public static void main(String[] args) {
+        final Scanner input = new Scanner(System.in);
+
+        System.out.println("Enter employee's name:");
+        final String employeeName = input.nextLine();
+
+        System.out.println("Enter number of hours worked in a week:");
+        final double hoursWorked = input.nextDouble();
+
+        System.out.println("Enter hourly pay rate:");
+        final double hourlyPayRate = input.nextDouble();
+
+        System.out.println("Enter federal tax withholding rate:");
+        final double federalTax = input.nextDouble();
+
+        System.out.println("Enter state tax withholding rate:");
+        final double stateTax = input.nextDouble();
+
+        final double grossPay = hourlyPayRate * hoursWorked;
+        final double federalWithholding = grossPay * federalTax;
+        final double stateWithholding = grossPay * stateTax;
+        final double totalDeduction = federalWithholding + stateWithholding;
+        final double netPay = grossPay - totalDeduction;
+
+
+        System.out.printf("Employee's Name: %s\nHours Worked: %.1f\nPay Rate: $%.2f\n" +

```

Comment:
Maybe you could make this one easier to read by breaking it up at new lines, such as:

~~~~Java
System.out.printf(
  "Employee's Name: %s\n"
  + "Hours Worked: %.1f\n"
  + "Pay Rate: $%.2f\n"
  + ...
);
~~~~